Incorrect behavior of ChangeArrayPushToArrayAssignRector

# Bug Report

<!-- First, thank you for reporting a bug. That takes time and we appreciate that! -->

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-master      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.org/demo/77ca42cb-c843-4fde-a45f-befd62bd03f3

```php
<?php
function force_balance_tags( $text ) {
	$tagstack  = array();
	$stacksize = 0;
	$tag = 'a';
	$stacksize = array_push( $tagstack, $tag );
}
```

### Responsible rules

* `ChangeArrayPushToArrayAssignRector`

## Expected Behavior

Rector should not apply this rule if the return value is going to be used.
Rector does shows this error when it's run:
```
[ERROR] Could not process  file, due to:                    
         "Node "PhpParser\Node\Expr\FuncCall":7 is child of "PhpParser\Node\Expr\Assign", so it cannot be removed as it would break PHP code. Change or remove the parent node instead.". 
```
The demo environment doesn't show this error and it would be nice if it did.
Also it'd be nice if errors like this would automattically skip rules so things could continue.

--- a/rules-tests/CodeQuality/Rector/FuncCall/ChangeArrayPushToArrayAssignRector/Fixture/skip_array_push_not_in_expression.php.inc
+++ b/rules-tests/CodeQuality/Rector/FuncCall/ChangeArrayPushToArrayAssignRector/Fixture/skip_array_push_not_in_expression.php.inc
@@ -1,0 +1,13 @@
+<?php
+
+namespace Rector\Tests\CodeQuality\Rector\FuncCall\ChangeArrayPushToArrayAssignRector\Fixture;
+
+final class SkipArrayPushNotInExpression
+{
+    public function run()
+    {
+        $tagstack  = array();
+        $tag = 'a';
+        $stacksize = array_push( $tagstack, $tag );
+    }
+}

--- a/rules/CodeQuality/Rector/FuncCall/ChangeArrayPushToArrayAssignRector.php
+++ b/rules/CodeQuality/Rector/FuncCall/ChangeArrayPushToArrayAssignRector.php
@@ -11,6 +11,7 @@ use PhpParser\Node\Expr\Assign;
 use PhpParser\Node\Expr\FuncCall;
 use PhpParser\Node\Stmt\Expression;
 use Rector\Core\Rector\AbstractRector;
+use Rector\NodeTypeResolver\Node\AttributeKey;
 use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
 use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
 
@@ -70,6 +71,11 @@ CODE_SAMPLE
         }
 
         if ($this->hasArraySpread($node)) {
+            return null;
+        }
+
+        $parent = $node->getAttribute(AttributeKey::PARENT_NODE);
+        if (! $parent instanceof Expression) {
             return null;
         }
 

